db_sql.go

```diff
@@ -304,14 +304,22 @@ func (x *sqlUserStoreDB) GetIdentities() ([]AuthUser, error) {
 }
 
 func (x *sqlUserStoreDB) GetUserFromIdentity(identity string) (AuthUser, error) {
-	return x.getUser(x.db.QueryRow("SELECT userid, email, username, firstname, lastname, mobile, authusertype FROM authuserstore WHERE (LOWER(email) = $1 AND (archived = false OR archived IS NULL)) OR (LOWER(username) = $1 AND (archived = false OR archived IS NULL))", CanonicalizeIdentity(identity)))
+	return getUserFromIdentity(x.db, identity)
 }
 
 func (x *sqlUserStoreDB) GetUserFromUserId(userId UserId) (AuthUser, error) {
-	return x.getUser(x.db.QueryRow("SELECT userid, email, username, firstname, lastname, mobile, authusertype FROM authuserstore WHERE userid = $1 AND (archived = false OR archived IS NULL)", userId))
+	return getUserFromUserId(x.db, userId)
 }
 
-func (x *sqlUserStoreDB) getUser(row *sql.Row) (AuthUser, error) {
+func getUserFromIdentity(db *sql.DB, identity string) (AuthUser, error) {
+	return getUser(db.QueryRow("SELECT userid, email, username, firstname, lastname, mobile, authusertype FROM authuserstore WHERE (LOWER(email) = $1 AND (archived = false OR archived IS NULL)) OR (LOWER(username) = $1 AND (archived = false OR archived IS NULL))", CanonicalizeIdentity(identity)))
```

Comment:
The "ARCHIVED" part of the logic can be hoisted out so that it is only expressed once.
`(a AND x) OR (b AND x) === (a OR b) AND x`